# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [5]:
# Hipotesis nula (H0), no hay diferencia de en la pérdida de peso entre las dos dietas
# Hipotesis alternativa (H1), hay diferencia en la pérdida de peso entre las dos dietas

from scipy.stats import ttest_ind

diet_1 = [2, 2.5, 3, 2.8, 2.3, 2.7, 2.5]
diet_2 = [3, 3.2, 3.1, 2.9, 2.8, 3, 3.2]

t_stat, p_value = ttest_ind(diet_1, diet_2)

print (f"El valor estadístico t = {t_stat}")
print (f"El valor de p = {p_value}")




El valor estadístico t = -3.5383407969933938
El valor de p = 0.004083270191713912


El valor de t negativo nos indica que la media entre el grupo de la dieta 1 es menor que la media entre el grupo de la dieta 2. El valor absoluto de t |3.54| podemos considerarlo grande, lo que indica una evidencia en contra de la hipótesis nula.

El valor de p = 0.004 si tenemos en cuenta que el grado de significancia alpha = 0.05, p_value es mucho mas pequeño que alpha, por lo que podemos rechazar la hipotésis nula (H0) planteada y decir que existe una diferencia en la péredida de peso promedio entre ambas dietas.

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [6]:
# Hipotesis nula (H0), la media de la producción de maíz de los tres fertilizantes es igual
# Hipotesis alternativa (H1), al menos una de las medias de producción de maíz de los fertilizantes es distinta

from scipy.stats import f_oneway

fertilizer_1 = [20, 21, 20, 19, 20]
fertilizer_2 = [22, 21, 23, 22, 21]
fertilizer_3 = [24, 23, 22, 23, 24]

f_stat, p_value = f_oneway(fertilizer_1, fertilizer_2, fertilizer_3)

print (f"El valor estadístico F = {f_stat}")
print (f"El valor de p = {p_value}")


EL valor estadístico F = 20.315789473684188
El valor de p = 0.00014047824793190475


El valor estadístico F (20.32), es alto, lo que sugiere que la variabilidad entre los grupos es mucho mayor que la variabilidad dentro de los grupos, lo que indica que al menos uno de los fertilizantes tiene un rendimiento diferente.

Como el valor de p = 0.00014 es mucho menos que el grado de significancia alpha = 0.05, podemos descartar la hipótesis nula, lo que indica que existe una gran diferencia en el rendimiento promedio entre los fertilizantes.

Tanto el valor de F como el valor de P se complementan indicando que de los 3 fertilizantes, uno de ellos tiene un rendimiento diferente.

La prueba ANOVA solo nos indica que hay una diferencia entre el rendimiento de los fertilizantes pero no nos indca cuál de llos es mejor. Para ello debemos realizar un análisis post hoc, por ejemplo, el test de Tukey. Dónde podremos comprobar cada par de fertilizantes y ver cuál tiene mayor rendimento.

In [7]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import numpy as np

data = np.array(fertilizer_1 + fertilizer_2 + fertilizer_3)
groups = ['Fertilizer 1'] * 5 + ['Fertilizer 2'] * 5 + ['Fertilizer 3'] * 5

tukey_results = pairwise_tukeyhsd(data, groups)
tukey_results.summary()


<class 'statsmodels.iolib.table.SimpleTable'>

Fertilizer 1 vs. Fertilizer 2: La diferencia de medias es de 1.8. El valor p (p-adj) es 0.0099, que es menor que 0.05. Esto indica que hay una diferencia  entre estos dos fertilizantes, lo que sugiere que el fertilizante 2 es mas efectivo que el fertilizante 1.

Fertilizer 1 vs. Fertilizer 3: La diferencia de medias es de 3.2, con un valor de p de 0.0001. De nuevo, este valor es menor que 0.05, lo que indica que el fertilizante 3 es más efectivo que el fertilizante 1.

Fertilizer 2 vs. Fertilizer 3: La diferencia de medias es de 1.4 y el valor de p es 0.0409. Al igual que en las comparaciones anteriores, este valor también es menor que 0.05, indicando que el fertilizante 3 es mas efectivo que el fertilizante 2.

Viendo las comparaciones entre los distintos fertilizantes podemos concluir que el fertilizante 3 es más efectivo que los otros dos.